pyresample/bucket/__init__.py

```diff
@@ -189,63 +223,40 @@ def _mask_bins_with_nan_if_not_skipna(self, skipna, data, out_size, statistic):
         data = data.ravel()
 
-        # Remove NaN values from the data when used as weights
-        weights = da.where(np.isnan(data), 0, data)
-
         # Rechunk indices to match the data chunking
-        if weights.chunks != self.idxs.chunks:
-            self.idxs = da.rechunk(self.idxs, weights.chunks)
+        if data.chunks != self.idxs.chunks:
+            self.idxs = da.rechunk(self.idxs, data.chunks)
 
         # Calculate the min of the data falling to each bin
         out_size = self.target_area.size
 
-        # merge into one Dataframe
-        df = dd.concat([dd.from_dask_array(self.idxs), dd.from_dask_array(weights)],
-                       axis=1)
-        df.columns = ['x', 'values']
+        # sort idxs and data
+        order = da.from_delayed(_sort_weights(statistic_method, data),
+                                shape=(len(data), ),
+                                dtype='int')
+        idxs_sorted = self.idxs[order]
+        data_sorted = np.append(data[order], np.nan)
 
-        if scipy_method == 'min':
-            statistics = df.map_partitions(lambda part: part.groupby(
-                                           np.digitize(part.x,
-                                                       bins=np.linspace(0, out_size, out_size)
-                                                       )
-                                           )['values'].min())
-
-        elif scipy_method == 'max':
-            statistics = df.map_partitions(lambda part: part.groupby(
-                                           np.digitize(part.x,
-                                                       bins=np.linspace(0, out_size, out_size)
-                                                       )
-                                           )['values'].max())
-
-        # fill missed index
-        statistics = (statistics + pd.Series(np.zeros(out_size))).fillna(0)
-
-        counts = self.get_sum(np.logical_not(np.isnan(data)).astype(np.int64)).ravel()
-
-        # TODO remove following line in favour of weights = data when dask histogram bug (issue #6935) is fixed
-        statistics = self._mask_bins_with_nan_if_not_skipna(skipna, data, out_size, statistics)
+        bins = np.linspace(0, out_size - 1, out_size).astype('int')
 
-        # set bin without data to fill value
-        statistics = da.where(counts == 0, fill_value, statistics)
+        statistics = da.from_delayed(_get_bin_statistic(bins, idxs_sorted, data_sorted),
+                                     shape=(len(bins),),
+                                     dtype=np.float64)
 
         return statistics.reshape(self.target_area.shape)
 
     def get_min(self, data, fill_value=np.nan, skipna=True):
         """Calculate minimums for each bin with drop-in-a-bucket resampling.
 
-        .. warning::
+        .. note::
 
-            The slow :meth:`pandas.DataFrame.groupby` method is temporarily used here,
+            The :meth:`numpy.ufunc.reduceat` method is used here,
```

Comment:
It seems that `numpy.ufunc.reduceat` isn't used anymore, so the note can be removed.